client/state/data-layer/wpcom/comments/index.js

```diff
@@ -188,6 +191,68 @@ export const announceDeleteFailure = ( action ) => {
 	];
 };
 
+export const emptyComments = ( action ) => ( dispatch ) => {
+	const { siteId, status } = action;
+
+	dispatch(
+		http(
+			{
+				apiVersion: '1',
+				body: {
+					empty_status: status,
+				},
+				method: 'POST',
+				path: `/sites/${ siteId }/comments/delete`,
+			},
+			{
+				...action,
+			}
+		)
+	);
+};
+
+export const handleEmptySuccess = (
+	{ status, siteId, options, refreshCommentListQuery },
+	apiResponse
+) => {
+	const showSuccessNotice = options?.showSuccessNotice;
+
+	return [
+		showSuccessNotice &&
+			successNotice(
+				status === 'spam' ? translate( 'Spam emptied.' ) : translate( 'Trash emptied.' ),
+				{
+					duration: DEFAULT_NOTICE_DURATION,
+					id: 'comment-notice',
+					isPersistent: true,
+				}
+			),
+		!! refreshCommentListQuery && requestCommentsList( refreshCommentListQuery ),
+		{
+			type: COMMENTS_EMPTY_RECEIVE,
+			siteId,
+			status,
+			commentIds: apiResponse?.results?.map( ( x ) => +x ), // convert to number
```

Comment:
In which scenario would `apiResponse` contain no results? Should we default to an empty array in this case?

```suggestion
			commentIds: apiResponse?.results?.map( ( x ) => +x ), // convert to number
```

Reply:
It turns out - never:

<img width="557" alt="Screenshot 2023-03-17 at 3 05 52 PM" src="https://user-images.githubusercontent.com/17325/225793636-cbd01c29-f7e5-457a-ac28-f452ddd5ddf6.png">

This was an API request where there were no comments to clean up.

So no need to be as defensive here.

Reply:
Nice 👍